### PR TITLE
Text box - On Enter key press event property (MX 8.7)

### PR DESCRIPTION
**Pull request by IIsaku**
- Add documentation for the introduced text box event property "On Enter key press" in Mendix 8.7
- Adjust descriptions for the already existing event properties.

### CODE REVIEW

**Comment by ConnorLand** [comment]
Properties initially encased in quotes here. This ties into larger question on property formatting which affects other docs. I will solve this and make all docs consistent.

**Comment by IIsaku** [comment]
@ConnorLand I changed the "On change" description a bit. For a text box "On change" is also triggered when there is a value change and the "Enter" key is pressed. This behavior could be the same for other widgets. Maybe it's good to check this out and alter the "On change" description for those widgets as well.

File: content/refguide/text-box.md [hunk 1]
```diff
@@ -63,24 +63,30 @@ Text box properties consist of the following sections:
...
 
-The on-enter property specifies an action that will be executed when the widget is entered, either by using the <kbd>Tab</kbd> key or by clicking it with the mouse.
+The On enter property specifies an action that will be executed when the widget is entered, either by using the <kbd>Tab</kbd> key or by clicking it with the mouse.
 
 {{% snippet file="refguide/events-section-link.md" %}}
 
 #### 2.5.3 On Leave
 
-The on-leave property specifies an action that will be executed when leaving the widget, either by using the <kbd>Tab</kbd> key or by clicking another widget.
+The On leave property specifies an action that will be executed when leaving the widget, either by using the <kbd>Tab</kbd> key or by clicking another widget.
```
Comment (ConnorLand) [inline content/refguide/text-box.md]: I removed the quotes around properties for now. I'll determine their final formatting fate with Roman.

**Comment by ConnorLand** [comment]
> @ConnorLand I changed the "On change" description a bit. For a text box "On change" is also triggered when there is a value change and the "Enter" key is pressed. This behavior could be the same for other widgets. Maybe it's good to check this out and alter the "On change" description for those widgets as well.

Ok. @r0b1n and I will adopt this into a project we are doing on property formatting.

File: content/refguide/text-box.md [hunk 2]
```diff
@@ -61,26 +61,32 @@ Text box properties consist of the following sections:
...
-#### 2.5.3 On Leave
+#### 2.5.3 On-Leave
 
 The on-leave property specifies an action that will be executed when leaving the widget, either by using the <kbd>Tab</kbd> key or by clicking another widget.
 
 This differs from the [On change](#on-change) property in that the event will always be triggered, even if the value has not been changed.
 
 {{% snippet file="refguide/events-section-link.md" %}}
 
+#### 2.5.4 On-Enter Key Press
```
Comment (ConnorLand) [inline content/refguide/text-box.md]: Is this all one property? Therefore is it on-enter-key-press?

**Comment by ConnorLand** [comment]
> @ConnorLand I changed the "On change" description a bit. For a text box "On change" is also triggered when there is a value change and the "Enter" key is pressed. This behavior could be the same for other widgets. Maybe it's good to check this out and alter the "On change" description for those widgets as well.

Could you please technically verify this widget behavior?  If it is the same for other widgets, I will alter the on change description for those widgets.

File: content/refguide/text-box.md [hunk 3]
```diff
@@ -61,26 +61,32 @@ Text box properties consist of the following sections:
...
-The on-leave property specifies an action that will be executed when leaving the widget, either by using the <kbd>Tab</kbd> key or by clicking another widget.
+The on leave event property specifies an action that will be executed when leaving the widget, either by using the <kbd>Tab</kbd> key or by clicking another widget.
 
 This differs from the [On change](#on-change) property in that the event will always be triggered, even if the value has not been changed.
 
 {{% snippet file="refguide/events-section-link.md" %}}
 
+#### 2.5.4 On Enter Key Press Event
+
+The on enter key press event property specifies an action that will be executed when the widget is entered and the <kbd>Enter</kbd> key is pressed. In a web application the widget will keep focus after the action gets executed.
```
Comment (Andries-Smit) [inline content/refguide/text-box.md]:
```suggestion
The on enter key press event property specifies an action that will be executed when the focus is inside the widget  and the <kbd>Enter</kbd> key is pressed. In a web application the widget will keep focus after the action gets executed.
```

**Pull request merged**